subsys/bluetooth/host/hci_core.c

```diff
@@ -923,9 +928,89 @@ static void le_update_private_addr(void)
 #endif
 }
 
+struct adv_id_check_data {
+	u8_t id;
+	bool adv_enabled;
+};
+
+#if !defined(CONFIG_BT_EXT_ADV)
```

Comment:
How come you've decided to use the negative condition when you anyway have an #else branch as well? At least I find it easier to read the conditionals when negations are only used when absolutely necessary.

Reply:
Actually, couldn't you just merge the two variants of `adv_id_check_func()` and use `IS_ENABLED()` inside of it?